server/apps/publish/publish_filters/filter_condition_tests.py

```diff
@@ -0,0 +1,275 @@
+# -*- coding: utf-8; -*-
+#
+# This file is part of Superdesk.
+#
+# Copyright 2013, 2014, 2015 Sourcefabric z.u. and contributors.
+#
+# For the full copyright and license information, please see the
+# AUTHORS and LICENSE files distributed with this source code, or
+# at https://www.sourcefabric.org/superdesk/license
+
+from superdesk.tests import TestCase
+from apps.publish.publish_filters.filter_condition import FilterConditionService
+from eve.utils import ParsedRequest
+import json
+import superdesk
+import re
+
+
+class FilterConditionTests(TestCase):
+
+    def setUp(self):
+        super().setUp()
+        self.req = ParsedRequest()
+        with self.app.app_context():
+            self.articles = [{'_id': '1', 'urgency': 1, 'headline': 'story', 'state': 'fetched'},
+                             {'_id': '2', 'headline': 'prtorque', 'state': 'fetched'},
+                             {'_id': '3', 'urgency': 3, 'state': 'fetched'},
+                             {'_id': '4', 'urgency': 4, 'state': 'fetched'},
+                             {'_id': '5', 'urgency': 2, 'state': 'fetched'},
+                             {'_id': '6', 'state': 'fetched'}]
+
+            self.app.data.insert('archive', self.articles)
+
+    def _setup_elastic_args(self, elastic_translation, search_type='filter'):
+        if search_type == 'keyword':
+            self.req.args = {'source': json.dumps({'query': {'bool': {'should': [elastic_translation]}}})}
+        elif search_type == 'not':
+            self.req.args = {'source': json.dumps({'query': {'bool': {'must_not': [elastic_translation]}}})}
+        elif search_type == 'filter':
+            self.req.args = {'source': json.dumps({'query': {
+                                                   'filtered': {
+                                                       'filter': {
+                                                           'bool': {
+                                                               'should': [elastic_translation]}}}}})}
+
+    def test_mongo_using_like_filter_complete_string(self):
+        f = FilterConditionService()
+        doc = {'field': 'headline', 'operator': 'like', 'value': 'story'}
+        query = f.get_mongo_query(doc)
+        with self.app.app_context():
+            docs = superdesk.get_resource_service('archive').\
+                get_from_mongo(req=self.req, lookup=query)
+            self.assertEquals(1, docs.count())
+            self.assertEquals('1', docs[0]['_id'])
+
+    def test_mongo_using_like_filter_partial_string(self):
+        f = FilterConditionService()
+        doc = {'field': 'headline', 'operator': 'like', 'value': 'tor'}
+        query = f.get_mongo_query(doc)
+        with self.app.app_context():
+            docs = superdesk.get_resource_service('archive').\
+                get_from_mongo(req=self.req, lookup=query)
+            doc_ids = [d['_id'] for d in docs]
+            self.assertEquals(2, docs.count())
```

Comment:
The `assertEquals()` alias has been [deprecated](https://docs.python.org/3.3/library/unittest.html#deprecated-aliases) in Python 3.3, better to replace it everywhere with `assertEqual()` to stay future-proof.